content/resources/crownlabs_ssh/crownlabs_ssh.md

```diff
@@ -0,0 +1,108 @@
+## Upload your key on CrownLabs
+
+Now open your page on **CrownLabs** and go to the [**Account page**](https://preprod.crownlabs.polito.it/app/account).
+
+<br>
+
+1. In the central box you will see a tab called **SSH key**. Click it to navigate to the **SSH keys** **management** panel.
+
+![image-20211112184900823](./img/1_1.png)
+
+<br>
+
+2. Now click on **Add SSH key** to open a popup that will allow you to upload your key.
+
+![image-20211112185058163](./img/1_2.png)
+
+<br>
+
+3. Paste your **Public key** (previously copied) inside the popup and add a **Name** to later identify it (whatever you want).
+
+![image-20211112190017165](./img/1_3.png)
+
+<br>
+
+4. Now click on **Save**, the popup will close and your **SSH public key** will be uploaded.
+
+![image-20211112190234270](./img/1_4.png)
+
+<br>
+
+## Connect to a VM from terminal
+
+<br>
+
+1. Create an instance of a VM in your [**CrownLabs Dashboard page**](https://preprod.crownlabs.polito.it/app).
+
+![image-20211112191418307](./img/2_1.png)
+
+<br>
+
+2. Now move to the **[Active page](https://preprod.crownlabs.polito.it/app/active)** and click on the **SSH** icon inside the row of the **VM** that you have just created. A popup with a shell command inside will be shown, **copy** that command.
+
+![image-20211112193803509](./img/2_2.png)
+
+<br>
+
+3. Now open a **terminal**, paste the previously copied command and run it.
+   A prompt asking if you trust the host you are trying to connect with will appear. Type **yes** and press **Enter**.
+   Then another prompt asking for a **password** might appear. Type **crownlabs** and press **Enter** to **login** inside your **VM**.
+
+![image-20211112194835635](./img/2_3.png)
+
+<br>
+
+## Conclusion
+
+Now you are able to log in your **VMs** using **SSH**. In case you want to access from different PCs you can add other keys following the same process.
+
+If you are interested in an advanced tutorial about **SSH key** you can check this [link](https://docs.gitlab.com/ee/ssh/). 
+
+
+
+## How it works
+
+What allows you to connect is the **bastion**, as shown in the figure below. First of all **SSH client** will try to connect to the bastion (using the option **-J** which specify a **ProxyJump**) and to authenticate using the previously generated **private key**. The **bastion** will check the **client**'s authentication attempt using the **public key** previously shared using the **CrownLabs Dashboard**.
+
+Now if the **bastion** has authenticated you correctly you will be redirect to your **VM**. Infact the syntax of the end of the command is the classical one **user@ipadress** where the user will be **crownlabs** (the default user on each **VM**) and the **ip** is your **VM's ip**. The **VM** will authenticate you again using your **public and private keys** as the bastion have done before. The **VM** might ask a password in addition (which is **crownlabs**)
```

Comment:
```suggestion
If authentication succeeds, you are then redirected to your **VM**. 
Indeed, the second part of the command is the standard **user@ipadress**, where **user** is **crownlabs** (the default user on each **VM**) and **ip** is your **VM's ip**. 
The **VM** will authenticate you again using your **SSH key-pair**, as the bastion previously did.
Alternatively, it might fallback to password authentication (which is **crownlabs**).
```